src/xb-machine.c

```diff
@@ -1559,17 +1562,28 @@ xb_machine_func_eq_cb(XbMachine *self,
 					  error);
 
 	/* TEXT:TEXT */
-	if (xb_opcode_cmp_str(&op1) && xb_opcode_cmp_str(&op2))
+	if (xb_opcode_cmp_str(&op1) && xb_opcode_cmp_str(&op2)) {
+		if (priv->debug_flags & XB_MACHINE_DEBUG_FLAG_SHOW_SLOW_PATH) {
+			gchar *str1 = xb_opcode_to_string(&op1);
```

Comment:
Either this is a leak, or `str1` should be `const`. Same below in a few places.

Reply:
Good catch, thanks, fixed.